test/extensions/filters/http/cache/cache_filter_test.cc

```diff
@@ -15,17 +15,18 @@ namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace Cache {
-namespace {
 
 class CacheFilterTest : public ::testing::Test {
 protected:
   // The filter has to be created as a shared_ptr to enable shared_from_this() which is used in the
   // cache callbacks.
-  CacheFilterSharedPtr makeFilter(HttpCache& cache) {
-    auto filter = std::make_shared<CacheFilter>(config_, /*stats_prefix=*/"", context_.scope(),
-                                                context_.timeSource(), cache);
+  CacheFilterSharedPtr makeFilter(HttpCachePtr cache) {
+    auto filter =
+        std::make_shared<CacheFilter>(/*stats_prefix=*/"", context_.scope(), context_.timeSource(),
+                                      CacheFilterConfig(simple_cache_, config_));
```

Comment:
If you fix the other reference-to-a-temporary-object at source rather than by consuming it in the constructor then you'll have the same problem here, and would need to make this CacheFilterConfig persistent for the life of the test.